src/server/routes/contract/transactions/paginateTransactionReceipts.ts

```diff
@@ -0,0 +1,176 @@
+  contractAddresses: Type.Optional(Type.Array(Type.String())),
+});
+
+const responseSchema = Type.Object({
+  result: Type.Object({
+    cursor: Type.Optional(Type.String()),
+    receipts: transactionReceiptsSchema,
+    status: Type.String(),
+  }),
+});
+
+responseSchema.example = {
+  result: {
+    cursor: "abcd-xyz",
+    receipts: [
+      {
+        chainId: 1,
+        blockNumber: 100,
+        contractAddress: "0x...",
+        transactionHash: "0x...",
+        blockHash: "0x...",
+        timestamp: 100,
+
+        to: "0x...",
+        from: "0x...",
+        transactionIndex: 1,
+
+        gasUsed: "1000",
+        effectiveGasPrice: "1000",
+        status: 1,
+      },
+    ],
+    status: "success",
+  },
+};
+
+const CursorSchema = z.object({
+  createdAt: z.number().transform((s) => new Date(s)),
+  chainId: z.number(),
+  blockNumber: z.number(),
+  transactionIndex: z.number(),
+});
+
+export async function pageTransactionReceipts(fastify: FastifyInstance) {
+  fastify.route<{
+    Reply: Static<typeof responseSchema>;
+    Querystring: Static<typeof requestQuerySchema>;
+  }>({
+    method: "GET",
+    url: "/contract/transactions/paginate-receipts",
+    schema: {
+      summary:
+        "Get contract paginated transaction receipts for subscribed contract",
+      description:
+        "Get contract paginated transaction receipts for subscribed contract",
+      tags: ["Contract-Transactions"],
+      operationId: "pageTransactionReceipts",
+      querystring: requestQuerySchema,
+      response: {
+        ...standardResponseSchema,
+        [StatusCodes.OK]: responseSchema,
+      },
+    },
+    handler: async (request, reply) => {
+      const { cursor, pageSize, contractAddresses } = request.query;
+
+      const standardizedContractAddresses = contractAddresses?.map((val) =>
+        val.toLowerCase(),
+      );
+
+      // add lag behind to account for clock skew, concurrent writes, etc
+      const config = await getConfiguration();
+      const maxCreatedAt = new Date(
+        Date.now() - config.cursorDelaySeconds * 1000,
+      );
+
+      let cursorObj;
+      try {
+        if (cursor) {
+          const decodedCursor = base64.decode(cursor);
+          const parsedCursor = decodedCursor
+            .split("-")
+            .map((val) => parseInt(val));
+          const [createdAt, chainId, blockNumber, transactionIndex] =
+            parsedCursor;
+          const validationResult = CursorSchema.safeParse({
+            createdAt,
+            chainId,
+            blockNumber,
+            transactionIndex,
+          });
+
+          if (!validationResult.success) {
+            throw new Error("Invalid cursor format");
+          }
+
+          cursorObj = validationResult.data;
+        }
+      } catch (error) {
+        throw createCustomError(
+          "Invalid cursor",
+          StatusCodes.BAD_REQUEST,
+          "BAD_REQUEST",
+        );
+      }
+
```

Comment:
can we put this logic inside `getTransactionReceiptsByCursor()` so that others using this function can use the logic